docs/intree-migrate.md

```diff
@@ -0,0 +1,237 @@
+feature is enabled, the in-tree volume requests (`kubernetes.io/rbd`) will be
+redirected to a corresponding CSI  (`rbd.csi.ceph.com`) Driver.
+
+- [Prerequisite](#prerequisite)
+- [Volumes provisioned before enabling CSI migration](#volumes-created-before-enabling-csi-migration)
+  - [Mount volume to a POD](#mount-volume-to-a-pod)
+  - [Unmount volume](#unmount-volume)
+  - [Delete volume](#delete-volume)
+- [Volumes provisioned after enabling CSI migration](#volumes-provisioned-after-enabling-csi-migration)
+  - [Create volume](#create-volume)
+  - [Mount volume to a POD](#mount-volume-to-a-pod)
+  - [Resize volume](#resize-volume)
+  - [Unmount volume](#unmount-volume)
+  - [Delete volume](#delete-volume)
+- [References](#additional-references)
+
+## Prerequisite
+
+For in-tree RBD migration to CSI driver to be supported for your kubernetes
+cluster, the kubernetes version running in your cluster should be >= v1.23. We
+also need sidecar controllers ( csi-provisioner and csi-resizer) which are
+compatible with the kubernetes version v1.23 to be available in your cluster.
+You can enable the migration with a couple of feature gates in your kubernetes
+cluster. These feature gates are alpha in kubernetes 1.23 release.
+
+  * `CSIMigrationRBD`: when enabled, it will redirect traffic from in-tree rbd
+  plugin ( kubernetes.io/rbd )  to CSI driver ( rbd.csi.ceph.com) , default
+  to `false` now.
+  * `IntreePluginRBDUnregister`: Disables the RBD in-tree driver
+
+The feature gates can be enabled by:
+
+- Adding the feature flag to the
+  kube-controller-manager `--feature-gates=CSIMigrationRBD=true`
+- Adding the feature flag to the kubelet config:
+
+Ex:
+
+```
+featureGates:
+ CSIMigrationRBD: true
+```
+
+As a Kubernetes cluster operator that administers storage, here are the
+prerequisites that you must complete before you attempt migration to the RBD CSI
+driver:
+
+- You must install the Ceph CSI driver (rbd.csi.ceph.com), v3.5.0 or above, into
+  your Kubernetes cluster.
+- Considering the `clusterID` field is a required parameter for CSI driver for
+  its operations, but in-tree StorageClass has `monitors` field as a required
+  parameter, a Kubernetes storage admin has to create a clusterID based on the
+  monitors hash ( ex:`#echo -n '<monitors_string>' | md5sum`) in the CSI config
+  map and keep the monitors under this clusterID configuration. -Also, if the
+  value of `adminId` in the in-tree Storageclass is different from value `admin`
+  , the `adminSecretName` mentioned in the in-tree Storageclass has to be
+  patched with the base64 value of the `adminId` parameter value, otherwise this
+  step can be skipped.
+
+In below examples, `fast-rbd` is in-tree storageclass with provisioner
+referencing in-tree provisioner `kubernetes.io/rbd`.
+
+```console
+$ kubectl describe sc fast-rbd |grep -i provisioner
+Provisioner:           kubernetes.io/rbd
+```
+
+## Volumes provisioned before enabling CSI migration
```

Comment:
why these steps are needed? just mention used to delete the application or scale down the application. we don't need to validate or show what happens with intree driver.